src/ORM/Association.php

```diff
@@ -629,9 +629,7 @@ public function deleteAll($conditions)
      */
     protected function _dispatchBeforeFind($query)
     {
-        $table = $this->target();
-        $options = $query->getOptions();
-        $table->dispatchEvent('Model.beforeFind', [$query, new \ArrayObject($options), false]);
+        $query->triggerBeforeFind();
```

Comment:
Correct me if i'm wrong, but it seems to me that an event is defined by its name ('Model.beforeFind'), subject (in this case : null) and data (the query itself). I don't think the change I introduce modifies any of them.